### ISSUE TITLE
Remove deprecated DryRunVisitor

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7919]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#7919|
|Type|task|
|Created|2015-05-29 14:58:02Z|
|Branch|https://github.com/twisted/twisted/tree/remove-dryrunvisitor-7919|

Deprecated since Twisted 13.0, it can go.

<details><summary>Searchable metadata</summary>

```
trac-id__7919 7919
type__task task
reporter__hawkowl hawkowl
priority__normal normal
milestone__ 
branch__branches_remove_dryrunvisitor_7919 branches/remove-dryrunvisitor-7919
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__ 
time__1432911482137511 1432911482137511
changetime__1432914103810618 1432914103810618
version__None None
owner__hawkowl hawkowl
cc__jml
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

(In [44969]) Branching to remove-dryrunvisitor-7919.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

Builders spun, please review.

**Comment by twisted-trac** [comment]
|<img alt="sigmavirus24's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_5"></a>|sigmavirus24 commented|
|-|-|

This looks good to me and everything's green, except py-select-gc.

**Comment by twisted-trac** [comment]
|<img alt="sigmavirus24's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|sigmavirus24 set owner to @hawkowl|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_7"></a>|@hawkowl set status to `closed`|
|-|-|

(In [44974]) Merge remove-dryrunvisitor-7919: Remove twisted.trial.runner.DryRunVisitor

Author: hawkowl
Reviewer: sigmavirus24
Fixes: [#7919](https://github.com/twisted/twisted/issues/7919)
